rs-react-app/src/components/layout/AppContainer.tsx

```diff
@@ -0,0 +1,31 @@
+import { Component } from 'react';
+import type { AppContainerProps } from '../../interfaces/interface';
+import Header from '../header/Header.tsx';
+import ErrorButton from '../error/ErrorButton';
+import ResultsContainer from '../main/ResultsContainer';
+import { ErrorBoundary } from '../error/Error';
+
+/**
+ * AppContainer component that provides the main layout structure
+ */
+class AppContainer extends Component<AppContainerProps> {
+  render() {
+    const { query, onInputChange, onSearch, results, loading } = this.props;
+
+    return (
+      <div className="app-container">
```

Comment:
Why don't you use CSS Modules to isolate styles from each other and keep all the needed CSS near it's component?